Jenkinsfile

```
@@ -1,30 +1,74 @@
 @Library('lisk-jenkins') _
 
-Makefile = 'Makefile.jenkins'
+LISK_CORE_HTTP_PORT = 4000
+LISK_CORE_WS_PORT = 5001
+MYSQL_PORT = 3306
+REDIS_PORT = 6381
 
-def waitForHttp() {
-	waitUntil {
-		script {
-			dir('./docker') {
-				def api_available = sh script: "make -f ${Makefile} ready", returnStatus: true
-				return (api_available == 0)
-			}
+def checkOpenPort(nPort) {
+	def result = sh script: "nc -z localhost ${nPort}", returnStatus: true
+	return (result == 0)
+}
+
+def runServiceIfMissing(svcName, path, nPort) {
+	if (checkOpenPort(nPort) == false) {
+		echo "${svcName} is not running, starting a new instance on port ${nPort}"
+		dir(path) { sh "make up" }
+		if (checkOpenPort(nPort) == false) { 
+			dir(path) { sh "make logs" }
+			currentBuild.result = "FAILURE"
+			throw new Exception("Failed to run ${svcName} instance")
 		}
 	}
 }
 
+def echoBanner(msg) {
+	echo '----------------------------------------------------------------------'
+	echo msg
+	echo '----------------------------------------------------------------------'
+}
+
+def checkHttp(url) {
+	def result = sh script: "curl -s -f -o /dev/null ${url}", returnStatus: true
+	return (result == 0)
+}
+
+def waitForHttp(url) {
+	waitUntil { script { return checkHttp(url) } }
+}
+
 pipeline {
 	agent { node { label 'lisk-service' } }
 	options {
 		timeout(time: 8, unit: 'MINUTES')
 	}
 	environment {
-		ENABLE_HTTP_API='http-version1,http-version1-compat,http-status,http-test,http-version2'
-		ENABLE_WS_API='rpc,rpc-v1,blockchain,rpc-test,rpc-v2'
+		ENABLE_HTTP_API='http-status,http-test,http-version2'
+		ENABLE_WS_API='blockchain,rpc-test,rpc-v2'
 	}
 	stages {
+		stage ('Run required services') {
+			steps {
+				script {
+					echoBanner(STAGE_NAME)
+					runServiceIfMissing('Lisk Core', './jenkins/lisk-core', LISK_CORE_WS_PORT)
+					runServiceIfMissing('MySQL', './jenkins/mysql', MYSQL_PORT)
+					runServiceIfMissing('Redis', './jenkins/redis', REDIS_PORT)
+
+					// Install PM2
+					nvm(getNodejsVersion()) {
+						sh 'npm i -g pm2'
+					}
+
+					// Show env exports
+					echo 'Printing all exports'
+					sh 'export'
```

```suggestion
					sh 'export -p'
```
was this meant for debugging maybe? if not your should be very careful not to expose secrets that way.